cluster/src/main/java/org/apache/iotdb/cluster/entity/raft/DataStateMachine.java

```diff
@@ -82,59 +77,71 @@ public DataStateMachine(String groupId, PeerId peerId) {
   @Override
   public void onApply(Iterator iterator) {
     while (iterator.hasNext()) {
-
-      Closure closure = null;
-      DataGroupNonQueryRequest request = null;
-      BasicResponse response = null;
+      final Closure closure = iterator.done();
       final ByteBuffer data = iterator.getData();
-      try {
-        request = SerializerManager.getSerializer(SerializerManager.Hessian2)
-            .deserialize(data.array(), DataGroupNonQueryRequest.class.getName());
-      } catch (final CodecException e) {
-        LOGGER.error("Fail to decode IncrementAndGetRequest", e);
-      }
-      if (iterator.done() != null) {
-        /** It's leader to apply task **/
-        closure = iterator.done();
-        response = ((ResponseClosure) closure).getResponse();
+
+      /** It's leader to apply task **/
+      if (closure != null) {
+        RaftTaskManager.getInstance().execute(() -> applySingleTask(closure, data));
```

Comment:
We need to ensure the commit sequence for RAFT tasks according to our offline talk.

Reply:
I approve，it's necessary.